Add hamming/instructions.append with exception raising details (#2564)
exercism/python: exercises/practice/hamming/.docs/instructions.append.md

# Implementation notes: exceptions

This exercise expects the code to raise exceptions in certain circumstances.

> Exceptions are usually raised when an operation cannot be completed for some

Reviewer: FWIW, general markdown style on Exercism is once sentence per line, not limited to a width. This makes reviews, edits, and diffs easier!

> reason and there is no value that can be returned which would be correct.

Reviewer: I think you meant: and there is no applicable value to return?

"which would be correct" is amtrigous. Either the act of no value returning is correct, or returning a value would be correct, or a correct value cannot be returned.

Author: Operation cannot be performed. Function usually returns a value. Any value returned would be incorrect. There exists no value where returning that value would be correct.

Maybe,
> Exceptions are usually raised when an operation cannot be completed for some reason and therefore there is no value to return. Any value returned would be an incorrect value.

Reviewer: Yeah much better!

> Instead, an exception is raised. When raising an exception, the tests expect

Reviewer: Doesn't seem to be an "instead" clause as the previous sentence already says there is no clause. `Therefore` might be more applicable.

Author: Changed.

that the exception class in instantiated and contains a message.

```python
# Avoid this. There is no message and the tests will fail:
raise ValueError

> # Do this. Provide a clear message with as much detail as possible:

Reviewer: ```suggestion
# Do this. Provide a clear and actionable message:
```

I don't think we always want to go into detail, but whatever we do, there must be "just" enough information to take an action. A large assssss error message which you can't solve isn't as good as a short, solvable, error. Right?

Author: Changed.
